### PR TITLE
[REF][14.0] shopinvader_multi_cart: Use typology

### PR DESCRIPTION
**Pull request by ivantodorovich**
Refactor `shopinvader_multi_cart` to make use of a new sale.order `typology`.
Without this, carts and stored carts where mixed and they weren't easily identified.

### CODE REVIEW

File: shopinvader_multi_cart/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Camptocamp (https://www.camptocamp.com).
```
Comment (simahawk) [inline shopinvader_multi_cart/models/sale_order.py]: copyright should be `Camptocamp SA`

File: shopinvader_multi_cart/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# @author Iván Todorovich <ivan.todorovich@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    typology = fields.Selection(
```
Comment (simahawk) [inline shopinvader_multi_cart/models/sale_order.py:11]: "saved" sounds better.

**Review by simahawk — CHANGES_REQUESTED** [review]
LG

File: shopinvader_multi_cart/models/sale_order.py (hunk 2 above)
Comment (ivantodorovich) [reply shopinvader_multi_cart/models/sale_order.py:11]: code updated ;)

File: shopinvader_multi_cart/services/carts.py [hunk 3]
```diff
@@ -37,41 +37,37 @@ def get(self, _id):
...
 
     # The following methods are 'private' and should be never NEVER be called
     # from the controller.
     # All params are trusted as they have been checked before
 
-    def _clear_session_cart(self):
-        return {
-            "data": {},
-            "store_cache": {"cart": {}},
-            "set_session": {"cart_id": 0},
```
Comment (simahawk) [inline shopinvader_multi_cart/services/carts.py:69]: you should probably return this to set the cart to the new cart (at least w/ Locomotive)

**Review by simahawk — COMMENTED** [review]
the base PR is getting merged

File: shopinvader_multi_cart/services/carts.py (hunk 3 above)
Comment (ivantodorovich) [reply shopinvader_multi_cart/services/carts.py:69]: AFAICS it's not necessary

On cart select (`saved` -> `cart`), this is handled here: 

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader_multi_cart/services/carts.py#L51

On cart save (`cart` -> `saved`), it's handled here:

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader_multi_cart/services/cart.py#L17


Basically the `_to_json` method will prepare the response locomotive needs:

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader/services/cart.py#L411-L424

**Comment by ivantodorovich** [comment]
rebased

**Review by simahawk — APPROVED** [review]
